autopush-common/src/db/bigtable/bigtable_client/mod.rs

```diff
@@ -782,24 +782,29 @@ impl BigtableDb {
     /// the tablename.
     ///
     pub async fn health_check(&mut self, metrics: Arc<StatsdClient>) -> DbResult<bool> {
-        let req = PingAndWarmRequest {
-            name: self.instance_name.clone(),
-            ..Default::default()
-        };
-        // PingAndWarmResponse does not implement a stream since it does not return data.
+        // It is recommended that we pick a random key to perform the health check. Selecting
+        // a single key for all health checks causes a "hot tablet" to arise. The `PingAndWarm`
+        // is intended to be used prior to large updates and is not recommended for use in
+        // health checks.
+        // This health check is to see if the database is present, the response is not important
+        // other than it does not return an error.
+        let random_uaid = Uuid::new_v4().simple().to_string();
+        let mut req = read_row_request(&self.table_name, &random_uaid);
+        let mut filter = data::RowFilter::default();
+        filter.set_block_all_filter(true);
+        req.set_filter(filter);
         let _r = retry_policy(RETRY_COUNT)
             .retry_if(
                 || async {
                     self.conn
-                        .ping_and_warm_opt(&req, call_opts(self.health_metadata.clone()))
+                        .read_rows_opt(&req, call_opts(self.health_metadata.clone()))
                 },
                 retryable_error(metrics.clone()),
             )
             .await
             .map_err(|e| DbError::General(format!("BigTable connectivity error: {:?}", e)))?;
 
-        // Since this should return no rows (with the row key set to a value that shouldn't exist)
-        // the first component of the tuple should be None.
+        // Since this may return no rows, the first component of the tuple should be None.
```

Comment:
This is an old comment about previous usage of `into_future` here returning a tuple, which we're not doing anymore

```suggestion
```